packages/web3-core-subscriptions/src/subscription.js

```diff
@@ -310,6 +309,7 @@ Subscription.prototype.subscribe = function() {
                     _this.emit('error', error);
                 }
             });
+            _this.emit('connected', result);
```

Comment:
I don't get it, why we have to move this emitter outside the block? The `connected` event just implies that client is connected and should not be related to `this.options.requestManager.send`. I think we should remove that emitter, connection state should be managed, some where else regardless of `.send` for the request manager.

Reply:
This moves the emission to be after the subscription gets added via `_this.options.requestManager.addSubscription`, so that it is present when attempting to unsubscribe

Perhaps this could be refactored as you suggest, but I'd like to refrain from making big unnecessary changes in 1.x (to keep from causing new bugs), and just keep this in mind for the rewrite